src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.Icu.cs

```diff
@@ -238,8 +238,41 @@ private static int IcuGetGeoId(string cultureName)
         private static int IcuGetDigitSubstitution(string cultureName)
         {
             Debug.Assert(!GlobalizationMode.UseNls);
-            int digitSubstitution = IcuLocaleData.GetLocaleDataNumericPart(cultureName, IcuLocaleDataParts.DigitSubstitution);
-            return digitSubstitution == -1 ? (int) DigitShapes.None : digitSubstitution;
+            int digitSubstitution = IcuLocaleData.GetLocaleDataNumericPart(cultureName, IcuLocaleDataParts.DigitSubstitutionOrListSeparator);
+            return digitSubstitution == -1 ? (int) DigitShapes.None : (digitSubstitution & 0x0000FFFF);
+        }
+
+        private static string IcuGetListSeparator(string? cultureName)
+        {
+            Debug.Assert(!GlobalizationMode.UseNls);
+            Debug.Assert(cultureName != null);
+
+            int separator = IcuLocaleData.GetLocaleDataNumericPart(cultureName, IcuLocaleDataParts.DigitSubstitutionOrListSeparator);
+            if (separator != -1)
+            {
+                switch (separator & 0xFFFF0000)
```

Comment:
Should this mask be in a named const in IcuLocaleData.cs, since it's strongly related to the values of the various XxSep named consts there?

Reply:
I already have this PR ported to the 5.0 release. How important to apply this comment there? I can add it to master though.

Reply:
master is fine.  I didn't get a chance to review that PR before it was merged.

Reply:
The servicing PR is not merged yet. It is here https://github.com/dotnet/runtime/pull/44823 just in case.

Reply:
> The servicing PR is not merged yet. It is here #44823 just in case.

@tarekgh if you add a commit here and then run the `/backport ...` comment, if there is a PR open, it will just push the new commits to that PR.

Reply:
Ok, I'll do that then. Thanks!

Reply:
This is done and the delta change also included in the 5.0 PR too as @safern thankfully pointed.